libraries/DataFlash/LogStructure.h

```diff
@@ -34,6 +36,93 @@ struct PACKED log_Format {
     char labels[64];
 };
 
+struct PACKED log_Unit {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    char unit[64]; // you know, this might be overkill...
+};
+
+struct PACKED log_Format_Multiplier {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    double multiplier;
+};
+
+struct PACKED log_Format_Units {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t format_type;
+    char units[16];
+    char multipliers[16];
+};
+
+struct UnitStructure {
+    const char ID;
+    const char unit[64];
+};
+
+struct MultiplierStructure {
+    const char ID;
+    const double multiplier;
+};
+
+// all units here should be base units
+// This does mean battery capacity is here as "amp*second"
+const struct UnitStructure log_Units[] = {
+    { '-', "" },    // no units e.g. Pi, or a string
+    { '?', "UNKNOWN" },    // Units which haven't been worked out yet....
+    { 'A', "amp" },
+    { 'd', "degrees" }, // of the angular variety, -180 to 180
+    { 'b', "bytes" },
+    { 'k', "degrees/second" },
```

Comment:
degrees per second

Reply:
As mentioned, these are NOT human-readable units.  These are to be interpreted by the client and transformed into preferred display units (most notably radians into degrees, I guess ;-) )

Reply:
Yes, I understand that, and I had this discussion wth the team in April.
The consensus was: use a long human readable with no symbold string. This way humans can read and machine can parse it. The GCS is then free to replace the big, long string with the abreviation like "°/s" if it wants to.

Again consistency is the key here, making Ardupilot parameters, mavlink message fields and dataflashlogs all use the same string makes a lot of sense in my book!

Reply:
Note: absolute string matching is a pretty poor form of "machine parsing"!

I think what this comes to is that I'll have to change the units to match what's in the other two places.  Every GCS will need to have a map from those  strings to actual units.  If we add a unit then the GCSs will have to update their maps.

I am curious, 'though.  Which bits of the mavlink message definitions allow you to automatically put units on message fields?  Certainly not the free-form text?!  AFAICS it must be the latter, but if so that means I can have "latitude" as a unit :-P